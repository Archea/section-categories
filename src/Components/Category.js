import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { withRouter } from 'react-router'

export class Category extends Component {
  render() {
    return (
      <div className="Category">
        <Link
          to={`/Category/${this.props.Category.id}`}
          style={{ textDecoration: 'none' }}>
          <div>{this.props.Category.name}</div>
        </Link>
      </div>
    )
  }

  _pickCategory = async () => {
    // ... you'll implement this
  }
}

export default withRouter(Category)
