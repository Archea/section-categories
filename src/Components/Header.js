import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { withRouter } from 'react-router'

class Header extends Component {
  render() {
    //forgive the inline stylyes. this just quick make it look more like sbc
    return (
      <div className="flex pa1 justify-between nowrap blue">
        <div className="flex flex-fixed white">
          <div className="fw7 mr1">Selection Tool</div>
          <Link to="/" className="ml1 no-underline white">
            Categories
          </Link>
          <Link to="/Assemblies" className="ml1 no-underline white">
            Assemblies
          </Link>
        </div>
      </div>
    )
  }
}

export default withRouter(Header)
