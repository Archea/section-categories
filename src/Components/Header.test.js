import React from 'react'
import ReactDOM from 'react-dom'
import { Category } from './Category'
import renderer from 'react-test-renderer'
import Header from './Header'
import { shallow, configure, mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { StaticRouter } from 'react-router'

configure({ adapter: new Adapter() })

describe('Header router', () => {
  test('should render', () => {
    const context = {}
    const wrapper = shallow(
      <StaticRouter location="/" context={context}>
        <Header />
      </StaticRouter>,
    )
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
