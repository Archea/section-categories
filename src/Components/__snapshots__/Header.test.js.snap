// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header router should render 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(Header) />
</Router>
`;
